uwuno/src/main/java/com/learning/uwuno/controllers/playerController.java

```diff
@@ -25,82 +27,70 @@
+            return newPlayer.getUid();
+        }
+        catch (NoSuchElementException e) {
             throw new errorNotFound();
         }
+        catch (JsonProcessingException e) {
+            throw new internalServerError();
+        }
     }
 
     // PUTS
     // Update player name in given room uid
     @PutMapping(value = "rooms/{uid}/players")
     public void updatePlayerName(@RequestBody String json, @PathVariable String uid)
             throws JsonMappingException, JsonProcessingException {
-        if (StringUtils.isNumeric(uid)) {
-            ObjectMapper mapper = new ObjectMapper();
-            try {
-                // TODO: Check if name already exists
-                room room = containerService.getRoom(Integer.parseInt(uid));
-                Map<String, String> map = mapper.readValue(json, new TypeReference<Map<String, String>>(){});
-                player player = room.getPlayer(Integer.parseInt(map.get("pid")));
-                player.setName(map.get("newName"));
-            }
-            catch (NoSuchElementException e) {
-                throw new errorNotFound();
-            }
-            catch (JsonProcessingException e) {
-                throw new internalServerError();
-            }
-        } else {
+        try {
+            // OPTIONAL TODO: Check if name already exists and append number to it
+            Map<String, String> map = requestUtil.parseJson(json);
+            room room = containerService.getRoom(uid);
+            player player = room.getPlayer(map.get("id"));
+            player.setName(map.get("name"));
+        }
+        catch (NoSuchElementException e) {
             throw new errorNotFound();
         }
+        catch (JsonProcessingException e) {
+            throw new internalServerError();
+        }
     }
 
     // DELETES
     @DeleteMapping(value = "rooms/{uid}/players")
     public void deletePlayer(@RequestBody String json, @PathVariable String uid)
             throws JsonMappingException, JsonProcessingException {
-        if (StringUtils.isNumeric(uid)) {
-            ObjectMapper mapper = new ObjectMapper();
-            try {
-                room room = containerService.getRoom(Integer.parseInt(uid));
-                Map<String, String> map = mapper.readValue(json, new TypeReference<Map<String, String>>(){});
-                room.deletePlayer(Integer.parseInt(map.get("pid")));
-            }
-            catch (NoSuchElementException e) {
-                throw new errorNotFound();
-            }
-            catch (JsonProcessingException e) {
-                throw new internalServerError();
-            }
+        try {
+            Map<String, String> map = requestUtil.parseJson(json);
+            room room = containerService.getRoom(uid);
+            room.deletePlayer(map.get("id"));
         }
-        else {
+        catch (NoSuchElementException e) {
             throw new errorNotFound();
         }
+        catch (JsonProcessingException e) {
```

Comment:
If we need to call this catch twice for the same set of code. The utils function loses most of it's point.
Recommend: Make parser class that returns an object not of map, but will have getters for whatever parameter that way we do not have to handle these exceptions again for JSON errors.

Reply:
That means we will need to know exactly what keys we're passing through in our JSON. The request body might be different each time depending on the class. We'll be mixing the getters of different classes and it might even be null. Unless we're making a parser class that specifically deals with each class.

Reply:
You can pass what you want into the parser get. ie. get("name").

Reply:
.. but that means we'll be creating a new map each time...?

Reply:
The created map can be saved in the object. Mapper can be a static for the class.

Reply:
Made into a parser class, but the exception is still needed.